### PR TITLE
feat: Display trends in report metrics

### LINKED ISSUE #4040: Display trends in the metrics
### Context

Right now, Chatwoot supports the following metrics

- Total number of Conversations
- Total number of incoming messages
- Total number of outgoing messages
- Average resolution time
- Average first response time
- Resolution count

These are absolute metrics for that particular time period, it doesn't show what has changed from the previous time period.

An example can be seen here.
Screenshot 2022-02-22 at 3 22 55 PM

#### API Changes
- [ ] Calculate the previous time period value along with the current time period value and return it in the summary API.

Now, the summary API returns the details as follows

```
{
  "conversations_count":133,
  "incoming_messages_count":370,
  "outgoing_messages_count":354,
  "avg_first_response_time":"4770.18802719496469575",
  "avg_resolution_time":"620946.011525262025",
  "resolutions_count":53
}
```

Change it to the following format.

```
{
  "current": {
    "conversations_count":10,
    "incoming_messages_count":20,
    "outgoing_messages_count":30,
    "avg_first_response_time":"20",
    "avg_resolution_time":"40",
    "resolutions_count":10
  },
  "previous: {
     "conversations_count":12,
     "incoming_messages_count":20,
     "outgoing_messages_count":30,
     "avg_first_response_time":"20",
     "avg_resolution_time":"40",
     "resolutions_count":10
  },
  "trends": {
     "conversation_count: 20,
     "incoming_messages_count":20,
     "outgoing_messages_count":30,
     "avg_first_response_time":"20",
     "avg_resolution_time":"40",
     "resolutions_count":10
  }
}
```

Trends should be returned in percentage values. The calculation is shown as follows 
```
change_in_value_as_percent = (current_value - prev_value)/pre_value * 100
```

#### Changes in the UI

Screenshot 2022-02-22 at 3 27 39 PM

Note: Show only 1 decimal place in the UI, use rounded logic.

### PR DESCRIPTION
**Pull request by aswindevps**
## Description

This change displays the metric trend in reports. So users can understand the percentage of change in the metric from the previous value.

Fixes #4040

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

https://user-images.githubusercontent.com/5848740/157606322-783cf338-1fef-485f-a762-cc7602823be2.mp4

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue [hunk 1]
```diff
@@ -167,6 +169,30 @@ export default {
...
+          return 0;
+        }
+        return Math.round(
+          ((this.accountSummary[metric_key] -
+            this.accountSummary.previous[metric_key]) /
+            this.accountSummary.previous[metric_key]) *
+            100
+        );
+      };
+    },
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue:184]: This logic seems to be repeated, can you please create a mixin?

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue [hunk 2]
```diff
@@ -167,6 +169,30 @@ export default {
...
+    displayMetric() {
+      return metric_key => {
+        if (metric_key === 'avg_first_response_time') {
+          return formatTime(this.accountSummary[metric_key]);
+        }
+        if (metric_key === 'avg_resolution_time') {
+          return formatTime(this.accountSummary[metric_key]);
+        }
+        return this.accountSummary[metric_key];
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue:193]: This also is repeated.

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue [hunk 3]
```diff
@@ -167,6 +169,30 @@ export default {
...
+        }
+        return Math.round(
+          ((this.accountSummary[metric_key] -
+            this.accountSummary.previous[metric_key]) /
+            this.accountSummary.previous[metric_key]) *
+            100
+        );
+      };
+    },
+    displayMetric() {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue:185]:
```js
    displayMetric() {
      return metric_key => {
        if (['avg_first_response_time', avg_resolution_time'].includes(metric_key)) {
          return formatTime(this.accountSummary[metric_key]);
        }
        return this.accountSummary[metric_key];
      };
    },

```

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue (hunk 1 above)
Comment (aswindevps) [reply app/javascript/dashboard/routes/dashboard/settings/reports/components/WootReports.vue]: Done

**Review by sojan-official — DISMISSED** [review]
LGTM

**Review by sojan-official — APPROVED** [review]

**Pull request merged**